### PR TITLE
DM-24972: Update Readme and switch to lock files

### PR DESCRIPTION
**Pull request by brianv0**
This updates README.md and describes the use of .lock files.

The packages files are updated but they are not guaranteed to be installable, and the use of them is deprecated, but it is kept for now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,20 +42,38 @@ This will likely cause significant changes to the environment, and should normal
...
+1. Add the name of the new package to install to the `bleed` file. Try
+   to choose an appropriate section, if possible.
+1. Instantiate the environment based on a `lock` file: `conda create --name pinned_<date> --file conda-<platform>-64.lock`
    * Use YYYYMMDD format for `<date>`
    * Use `osx` or `linux` for `<platform>`
 1. Activate the environment: `source activate pinned_<date>`
 1. Install the package: `conda install --no-update-deps <packagename>`
-1. Export the environment, over-writing the previous `packages` file: `conda env export > conda3_packages-<ARCH>-64.yml`
-1. Update the `requirements.txt` file in the [linting repository](https://github.com/lsst/linting) to match the version numbers specified in the new conda environment.
+1. Export the environment, over-writing the previous `packages` file: `conda list --explicit > conda-<platform>-64.lock`
```
Comment (ktlim) [inline README.md]:
```suggestion
1. Export the environment, over-writing the previous `lock` file: `conda list --explicit > conda-<platform>-64.lock`
```

File: README.md [hunk 2]
```diff
@@ -36,20 +42,38 @@ This will likely cause significant changes to the environment, and should normal
    * Use `osx` or `linux` for `<platform>`
 1. Activate the environment: `source activate pinned_<date>`
 1. Check that the list of packages in the environment are as expected
-1. Export the environment, over-writing the previous `packages` file: `conda env export > conda3_packages-<ARCH>-64.yml`
-1. Update the `requirements.txt` file in the [linting repository](https://github.com/lsst/linting) to match the version numbers specified in the new conda environment.
```
Comment (ktlim) [inline README.md:40]: Doesn't this still hold, for whatever is listed there? (Only flake8 today, but maybe more later.)

**Review by ktlim — APPROVED** [review]
I was kind of thinking that the packages file could be kept as direct-dependency only, build-spec-free, but synchronized with the lock file.  But it doesn't matter much, so this is OK.

**Review by jdswinbank — APPROVED** [review]
Thanks for updating the README!

Only comment would be that it's probably worth maintaining the existing one-sentence-per-line style. Otherwise, this looks fine to me. 👍

**Pull request merged**